packages/webpack/src/plugins/webpack-nx-build-coordination-plugin.ts

```diff
@@ -56,44 +45,55 @@ export class WebpackNxBuildCoordinationPlugin {
     this.currentlyRunning = 'nx-build';
     return new Promise<void>((res) => {
       try {
-        const cp = exec(this.buildCmd);
+        this.buildCmdProcess = exec(this.buildCmd);
 
-        cp.stdout.pipe(process.stdout);
-        cp.stderr.pipe(process.stderr);
-        cp.on('exit', () => {
+        this.buildCmdProcess.stdout.pipe(process.stdout);
+        this.buildCmdProcess.stderr.pipe(process.stderr);
+        this.buildCmdProcess.on('exit', () => {
           res();
         });
-        cp.on('error', () => {
+        this.buildCmdProcess.on('error', () => {
           res();
         });
         // eslint-disable-next-line no-empty
       } catch (e) {
         res();
-      } finally {
-        this.currentlyRunning = 'none';
       }
+    }).then(() => {
+      this.currentlyRunning = 'none';
+      this.buildCmdProcess = null;
```

Comment:
This ensures webpack will wait until the build of dependent projects has completed before continuing the webpack build